lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -538,6 +540,33 @@ func makeSubscriberStateHandler(desiredState string) echo.HandlerFunc {
 	}
 }
 
+func getStatesForIMSIs(networkID string, typeFilter []string, keyPrefix *string, serdes serde.Registry) (state_types.StatesByID, error) {
+	// Post-filter functionality to only return states with the exact IMSI
+	// prefix as specified
+	states, err := state.SearchStates(networkID, typeFilter, nil, keyPrefix, serdes)
+	if err != nil {
+		return nil, err
+	}
+
+	for stateID := range states {
+		imsi := stateID.DeviceID
+		if stateID.Type == lte.MobilitydStateType {
+			matches := mobilitydStateKeyRe.FindStringSubmatch(stateID.DeviceID)
+			if len(matches) != mobilitydStateExpectedMatchCount {
+				glog.Infof("state device ID '%s' did not match IMSI-prefixed regex", stateID.DeviceID)
```

Comment:
If we want to log something here, might be useful to log the state type as well.

Reply:
^